cli/file_fetcher.rs

```diff
@@ -2,677 +2,499 @@
     redirect_limit: i64,
-  ) -> Result<Option<SourceFile>, AnyError> {
+  ) -> Result<Option<File>, AnyError> {
+    debug!("FileFetcher::fetch_cached - specifier: {}", specifier);
     if redirect_limit < 0 {
-      return Err(custom_error("Http", "too many redirects"));
+      return Err(custom_error("Http", "Too many redirects."));
     }
 
-    let result = self.http_cache.get(&module_url);
-    let result = match result {
-      Err(e) => {
-        if let Some(e) = e.downcast_ref::<std::io::Error>() {
-          if e.kind() == std::io::ErrorKind::NotFound {
-            return Ok(None);
+    let (mut source_file, headers) =
+      match self.http_cache.get(specifier.as_url()) {
+        Err(err) => {
+          if let Some(err) = err.downcast_ref::<std::io::Error>() {
+            if err.kind() == std::io::ErrorKind::NotFound {
+              return Ok(None);
+            }
           }
+          return Err(err);
         }
-        return Err(e);
-      }
-      Ok(c) => c,
-    };
-
-    let (mut source_file, headers) = result;
-    if let Some(redirect_to) = headers.get("location") {
-      let redirect_url = match Url::parse(redirect_to) {
-        Ok(redirect_url) => redirect_url,
-        Err(url::ParseError::RelativeUrlWithoutBase) => {
-          let mut url = module_url.clone();
-          url.set_path(redirect_to);
-          url
-        }
-        Err(e) => {
-          return Err(e.into());
-        }
+        Ok(cache) => cache,
       };
-      return self
-        .fetch_cached_remote_source(&redirect_url, redirect_limit - 1);
+    if let Some(redirect_to) = headers.get("location") {
+      let redirect =
+        ModuleSpecifier::resolve_import(redirect_to, specifier.as_str())?;
+      return self.fetch_cached(&redirect, redirect_limit - 1);
     }
+    let mut bytes = Vec::new();
+    source_file.read_to_end(&mut bytes)?;
+    let file = self.get_file(specifier, bytes, &headers)?;
 
-    let mut source_code = Vec::new();
-    source_file.read_to_end(&mut source_code)?;
-
-    let cache_filename = self.http_cache.get_cache_filename(module_url);
-    let fake_filepath = PathBuf::from(module_url.path());
-    let (media_type, charset) = map_content_type(
-      &fake_filepath,
-      headers.get("content-type").map(|e| e.as_str()),
-    );
-    let source_code = if let Some(charset) = charset {
-      text_encoding::convert_to_utf8(&source_code, &charset)?.to_string()
-    } else {
-      String::from_utf8(source_code)?
-    };
-    let types_header = headers.get("x-typescript-types").map(|e| e.to_string());
-    Ok(Some(SourceFile {
-      url: module_url.clone(),
-      filename: cache_filename,
-      media_type,
-      source_code,
-      types_header,
-    }))
+    Ok(Some(file))
   }
 
-  /// Asynchronously fetch remote source file specified by the URL following redirects.
-  ///
-  /// Note that this is a recursive method so it can't be "async", but rather return
-  /// Pin<Box<..>>.
```

Comment:
Why remove this comment?

Reply:
Accidental when rewriting the module... will restore.